Plugins/DD4hep/src/ConvertDD4hepDetector.cpp

```diff
@@ -548,79 +589,61 @@ void collectCompounds_dd4hep(dd4hep::DetElement& detElement,
   const dd4hep::DetElement::Children& children = detElement.children();
   for (auto& child : children) {
     dd4hep::DetElement childDetElement = child.second;
-    Acts::ActsExtension* detExtension = nullptr;
-    try {
-      detExtension = childDetElement.extension<Acts::ActsExtension>();
-    } catch (std::runtime_error& e) {
-    }
-    if ((detExtension != nullptr) &&
-        (detExtension->hasType("barrel", "detector") ||
-         detExtension->hasType("endcap", "detector"))) {
+    dd4hep::DetType type{childDetElement.typeFlag()};
+    if (type.is(dd4hep::DetType::BARREL) || type.is(dd4hep::DetType::ENDCAP)) {
       compounds.push_back(childDetElement);
-      continue;
     }
     collectCompounds_dd4hep(childDetElement, compounds);
   }
 }
 
 void collectSubDetectors_dd4hep(dd4hep::DetElement& detElement,
-                                std::vector<dd4hep::DetElement>& subdetectors) {
+                                std::vector<dd4hep::DetElement>& subdetectors,
+                                LoggerWrapper logger) {
   const dd4hep::DetElement::Children& children = detElement.children();
   for (auto& child : children) {
     dd4hep::DetElement childDetElement = child.second;
-    Acts::ActsExtension* detExtension = nullptr;
-    try {
-      detExtension = childDetElement.extension<Acts::ActsExtension>();
-    } catch (std::runtime_error& e) {
-      if (childDetElement.type() == "compound") {
-        subdetectors.push_back(childDetElement);
-        continue;
-      }
-    }
-    if ((detExtension != nullptr) &&
-        (detExtension->hasType("barrel", "detector") ||
-         detExtension->hasType("beampipe", "layer") ||
-         detExtension->hasType("passive cylinder", "layer"))) {
+    dd4hep::DetType type{childDetElement.typeFlag()};
+    if (childDetElement.type() == "compound") {
       subdetectors.push_back(childDetElement);
       continue;
     }
-    collectSubDetectors_dd4hep(childDetElement, subdetectors);
+
+    if (type.is(dd4hep::DetType::TRACKER)) {
+      subdetectors.push_back(childDetElement);
+    }
+    collectSubDetectors_dd4hep(childDetElement, subdetectors, logger);
   }
 }
 
 void collectLayers_dd4hep(dd4hep::DetElement& detElement,
```

Comment:
Not sure why we hadn't used our own naming conventions here.

Reply:
🤷